src/crafting.cpp

```diff
@@ -24,6 +24,28 @@
 #include <math.h>    //sqrt
 #include <algorithm> //std::min
 
+enum usage {
+	use_from_map = 1,
+	use_from_player = 2,
+	use_from_both = 1 | 2,
+	use_from_none = 4,
+	cancel = 5
+};
+
+struct comp_selection {
```

Comment:
I thought about doing this, but was unsure about how to add type constraints in c++.